labgraph/lsl_node/lsl_poller_node.py

```diff
@@ -0,0 +1,37 @@
+from pylsl import StreamInlet, resolve_stream
+from ..graphs import AsyncPublisher, Config, Node, Topic, publisher
+from ..util.logger import get_logger
+from ..lsl_node import LSLMessage
+import asyncio
+
+
+logger = get_logger(__name__)
+
+
+class LSLPollerConfig(Config):
+    type: str = 'labgraph'
+
+
+class LSLPollerNode(Node):
+    """
+    represents a node in the graph which recieves data from an LSL
+    stream, data received is subsequently pushed to rest of the graph
+    as as LSLMessage
+    Args:
+        type: The name the type assigned to incoming stream
+        (default 'labgraph')
+    """
+    topic = Topic(LSLMessage)
+    config: LSLPollerConfig
+
+    def setup(self) -> None:
+        logger.log(1, f"looking for an {self.config.type} Stream")
+        self.streams = resolve_stream('type', self.config.type)
+        self.inlet = StreamInlet(self.streams[0])
+
+    @publisher(topic)
+    async def lsl_subscriber(self) -> AsyncPublisher:
+        while True:
+            sample, timestamp = self.inlet.pull_sample()
+            yield self.topic, LSLMessage(sample)
+            await asyncio.sleep(0.1)
```

Comment:
Please change the constant to a variable.

Reply:
Done